chainer/exporters/caffe.py

```diff
@@ -0,0 +1,428 @@
+                for k, v in six.iteritems(concat_param):
+                    setattr(layer.concat_param, k, v)
+
+        elif func.label == 'Softmax':
+            params['type'] = 'Softmax'
+
+        elif func.label == 'Reshape':
+            input_ = func.inputs[0]
+            parent = input_.creator
+            parent_layer_name = parent_layer_names[0]
+            if 'Reshape' in parent_layer_name:
+                grandparent = parent.inputs[0].creator
+                parent_layer_name = self._get_parent_name(grandparent)
+            reshape_param = {'shape': {'dim': list(func.shape)}}
+            params['type'] = 'Reshape'
+            params['bottom'] = [parent_layer_name]
+            params['reshape_param'] = reshape_param
+            if layer is not None:
+                dim = reshape_param['shape']['dim']
+                layer.reshape_param.shape.dim[:] = dim
+
+        elif func.label == '_ + _':
+            params['type'] = 'Eltwise'
+
+        else:
+            raise Exception(
+                'Cannot convert, name={}, rank={}, label={}, inputs={}'.format(
+                    layer_name, func.rank, func.label, parent_layer_names))
+        if prototxt is not None:
+            prototxt.write(self._gen_layer_prototxt(params))
+
+        if net is not None:
+            layer.name = params['name']
+            layer.type = params['type']
+            layer.bottom[:] = params['bottom']
+            layer.top[:] = params['top']
+            layer.phase = caffe_pb.TEST
+
+    def __call__(self, name, inputs, outputs):
+        dumped_list = _dump_graph(outputs)
+        f = None
+        net = None
+        if self.caffemodel is not None:
+            net = caffe_pb.NetParameter()
+        try:
+            if self.prototxt is not None:
+                f = open(self.prototxt, 'wt')
+                f.write('name: "{}"\n'.format(name))
+                assert len(inputs) == 1
+                f.write('layer {\n'
+                        '  name: "data"\n'
+                        '  type: "Input"\n'
+                        '  top: "data"\n'
+                        '  input_param { shape: {')
+                for i in inputs[0].shape:
+                    f.write(' dim: ' + str(i))
+                f.write(' } }\n'
+                        '} \n')
+            for i in dumped_list:
+                self.dump_function_object(i, f, net)
+        finally:
+            if f is not None:
+                f.close()
+
+        if net is not None:
+            with open(self.caffemodel, 'wb') as f:
+                f.write(net.SerializeToString())
+            if self.debug:
+                import google.protobuf.text_format
+                with open(self.caffemodel + ".txt", 'w') as f:
+                    f.write(google.protobuf.text_format.MessageToString(net))
+
+
+def export(model, args, directory=None,
+           export_params=True, graph_name='Graph'):
+    """Export computational graph as Caffe format.
+
```

Comment:
As I stated before, please write down the list of supported functions that can be included in the given model and some comments of what users should be noted if there's any restrictions or limitations here. And it's also better to describe what architectures are tested.